x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/enrollment/CreateEnrollmentTokenTests.java

```diff
@@ -0,0 +1,312 @@
+
+        final HttpResponse httpResponseNotOK = new HttpResponse(HttpURLConnection.HTTP_BAD_REQUEST, new HashMap<String, Object>());
+        when(client.execute(anyString(), eq(createAPIKeyURL), anyString(), any(SecureString.class), any(CheckedSupplier.class),
+            any(CheckedFunction.class))).thenReturn(httpResponseNotOK);
+
+        IllegalStateException ex = expectThrows(IllegalStateException.class, () ->
+            createEnrollmentToken.createNodeEnrollmentToken("elastic", new SecureString("elastic")));
+        assertThat(ex.getMessage(), Matchers.containsString("Unexpected response code [400] from calling POST "));
+    }
+
+    public void testFailedRetrieveHttpInfo() throws Exception {
+        final CommandLineHttpClient client = mock(CommandLineHttpClient.class);
+        when(client.getDefaultURL()).thenReturn("http://localhost:9200");
+        final CreateEnrollmentToken createEnrollmentToken = new CreateEnrollmentToken(environment, client);
+        final URL createAPIKeyURL = createEnrollmentToken.createAPIKeyUrl();
+        final URL getHttpInfoURL = createEnrollmentToken.getHttpInfoUrl();
+
+        final HttpResponse httpResponseOK = new HttpResponse(HttpURLConnection.HTTP_OK, new HashMap<String, Object>());
+        when(client.execute(anyString(), eq(createAPIKeyURL), anyString(), any(SecureString.class), any(CheckedSupplier.class),
+            any(CheckedFunction.class))).thenReturn(httpResponseOK);
+
+        String createApiKeyResponseBody;
+        try (XContentBuilder builder = XContentFactory.contentBuilder(XContentType.JSON)) {
+            builder.startObject()
+                .field("id", "DR6CzXkBDf8amV_48yYX")
+                .field("name", "enrollment_token_API_key_VuaCfGcBCdbkQm")
+                .field("expiration", "1622652381786")
+                .field("api_key", "x3YqU_rqQwm-ESrkExcnOg")
+                .endObject();
+            createApiKeyResponseBody = Strings.toString(builder);
+        }
+        when(client.execute(eq("POST"), eq(createAPIKeyURL), eq(ElasticUser.NAME), any(SecureString.class),
+            any(CheckedSupplier.class), any(CheckedFunction.class)))
+            .thenReturn(createHttpResponse(HttpURLConnection.HTTP_OK, createApiKeyResponseBody));
+
+        final HttpResponse httpResponseNotOK = new HttpResponse(HttpURLConnection.HTTP_BAD_REQUEST, new HashMap<String, Object>());
+        when(client.execute(anyString(), eq(getHttpInfoURL), anyString(), any(SecureString.class), any(CheckedSupplier.class),
+            any(CheckedFunction.class))).thenReturn(httpResponseNotOK);
+
+        IllegalStateException ex = expectThrows(IllegalStateException.class, () ->
+            createEnrollmentToken.createNodeEnrollmentToken("elastic", new SecureString("elastic")));
+        assertThat(ex.getMessage(), Matchers.containsString("Unexpected response code [400] from calling GET "));
+    }
+
+    public void testNoKeyStore() throws Exception{
```

Comment:
Since we're doing PrivateKeyEntry filtering in this class, I think we should add a test with a keystore with many CA entries and a keystore with no CA entries too.